src/Core/src/AbstractApi.php

```diff
@@ -83,11 +84,29 @@ public function __construct($configuration = [], ?CredentialProvider $credential
         return $request->getEndpoint();
     }
 
-    abstract protected function getServiceCode(): string;
+    /**
+     * @deprecated
+     */
+    protected function getServiceCode(): string
+    {
+        throw new LogicException(\sprintf('The method "%s" should not be called. The Client "%s" must implement the "%s" method.', __FUNCTION__, \get_class($this), 'getEndpointMetadata'));
+    }
 
-    abstract protected function getSignatureVersion(): string;
+    /**
+     * @deprecated
+     */
+    protected function getSignatureVersion(): string
```

Comment:
We can just remove these and update the version constraints. Right?

Reply:
yes, in all but 1.0 clients (because they may have been extends right?)